### PR TITLE
occt: Simplify static linking

### PR DESCRIPTION
**Pull request by mwestphal**

### CODE REVIEW

File: plugins/occt/module/CMakeLists.txt [hunk 1]
```diff
@@ -16,9 +16,12 @@ vtk_module_include(f3d::vtkextOCCT
   PRIVATE
     $<BUILD_INTERFACE:${OpenCASCADE_INCLUDE_DIR}>)
 
-vtk_module_link(f3d::vtkextOCCT
-  PRIVATE
-    ${OpenCASCADE_LIBRARIES})
+set(_occt_link_libs "${OpenCASCADE_LIBRARIES}")
+if(CMAKE_LINK_GROUP_USING_RESCAN_SUPPORTED)
```
Comment (Meakk) [inline plugins/occt/module/CMakeLists.txt:20]: change the minimum CMake version for `occt` plugin to 3.24

**Review by Meakk — CHANGES_REQUESTED** [review]

File: plugins/occt/module/CMakeLists.txt (hunk 1 above)
Comment (mwestphal) [reply plugins/occt/module/CMakeLists.txt:20]: I dont think we want to do that. 3.24 is quite new, it is not strictly required to build the occt plugin.

Comment (mwestphal) [reply plugins/occt/module/CMakeLists.txt:20]: And in anycase that check is needed, because rescan is a GCC linker thing only

Comment (Meakk) [reply plugins/occt/module/CMakeLists.txt:20]: ok but then `CMAKE_LINK_GROUP_USING_RESCAN_SUPPORTED` won't be defined

Comment (Meakk) [reply plugins/occt/module/CMakeLists.txt:20]: > ok but then `CMAKE_LINK_GROUP_USING_RESCAN_SUPPORTED` won't be defined

That would still work if undefined, but I believe we are checking if variables are undefined?

Comment (mwestphal) [reply plugins/occt/module/CMakeLists.txt:20]: `if(VAR)` is not warning if VAR is not defined.

Comment (Meakk) [reply plugins/occt/module/CMakeLists.txt:20]: Ok then

**Pull request merged**